drivers/spi/spi_mcux_flexcomm.c

```diff
@@ -771,17 +806,23 @@ static int spi_mcux_release(const struct device *dev,
 {
 	struct spi_mcux_data *data = dev->data;
 
+	pm_policy_device_power_lock_get(data->dev);
+
 	spi_context_unlock_unconditionally(&data->ctx);
 
+	pm_policy_device_power_lock_put(dev);
+
```

Comment:
i dont think you need to lock here, this is only doing software state manipulation, not changing hardware state